components/ModalDialog.vue

```diff
@@ -0,0 +1,41 @@
+<template>
+  <div
+    class="fixed top-0 right-0 bottom-0 left-0 flex justify-center bg-gray-900 bg-opacity-80"
+    @click="$emit('close-modal')"
+  >
+    <div
+      class="justify-center text-left overflow-auto m-10 p-10 rounded-xl mt-10 bg-white prose"
+      @click.stop
+    >
+      <h2>{{ dataset.title }}</h2>
+      <ul>
+        <li>Contact: {{ dataset.contact }}</li>
```

Comment:
@Peter9192 Got some trouble here. When referring the first instance in the array as `dataset.contact[0].name`, there will be an error which indicates the array is not rendered. But with `console.log()` it shows the array is there. The `v-if` trick is also not working. Any idea about it? I have a feeling that it is due to the promise thingy.

Reply:
@Peter9192 suggests to solve it using the `asyncData`, which ensures that the data will be fetched before the rendering of the page, which works perfectly! Thanks for the tip!